File: 22.08/35.go
func searchInsert(nums []int, target int) int {
    s := 0
    e := len(nums)
    
    for s < e {
        mid := (s+e)/2
        if nums[mid] > target {
            e = mid
        } else if nums[mid] == target {
            return mid
        }else {
            s = mid + 1
        }
    }
    
    return s
    
}

File: typescript/23.08/35.ts
function searchInsert(nums: number[], target: number): number {
    let [low, mid, top] = [0, 0, nums.length]

    while (low < top) {
        mid = Math.floor((low + top)/2)
        if (nums[mid] === target) {
            return mid
        }
        if (nums[mid] < target) {
            low = mid + 1
        } else {
            top = mid
        }
    }
    return low
};
